xrpl/asyncio/transaction/main.py

```diff
@@ -75,18 +76,39 @@ async def sign(
         wallet: the wallet with which to sign the transaction.
         check_fee: whether to check if the fee is higher than the expected transaction
             type fee. Defaults to True.
+        multisign: whether to sign the transaction for a multisignature transaction.
 
     Returns:
         The signed transaction blob.
     """
-    if check_fee:
-        await _check_fee(transaction)
-    transaction_json = _prepare_transaction(transaction, wallet)
-    serialized_for_signing = encode_for_signing(transaction_json)
-    serialized_bytes = bytes.fromhex(serialized_for_signing)
-    signature = keypairs_sign(serialized_bytes, wallet.private_key)
-    transaction_json["TxnSignature"] = signature
-    return Transaction.from_xrpl(transaction_json)
+    if not multisign:
```

Comment:
This if statement should be flipped - the logic becomes minorly complicated to understand when the `else` is called (`else if not not multisign`), so it's better to just have an `if multisign` block and flip the if and else blocks.